File: Framework/LanguageSupport/cpp/MMICPP/Adapter/AvatarContent.cpp
// SPDX-License-Identifier: MIT
// The content of this file has been developed in the context of the MOSIM research project.
// Original author(s): Andreas Kaiser, Niclas Delfs, Stephan Adam

#include "AvatarContent.h"

AvatarContent::AvatarContent(string avatarId) :avatarID{ avatarId }
{
}

 MotionModelUnitBaseIf & AvatarContent::GetMMUbyId(const string &mmuId) const
{
	auto iter = this->MMUs.find(mmuId);
	if (iter != this->MMUs.end()) // mmu found!
	{
		return *iter->second;
	}
	else
	{
		throw runtime_error("MMU with id: " + mmuId + " not found in avatar content");
	}
}

File: Framework/LanguageSupport/java/MMIJava/src/Adapter/AvatarContent.java
// SPDX-License-Identifier: MIT
// The content of this file has been developed in the context of the MOSIM research project.
// Original author(s): Andreas Kaiser, Felix Gaisbauer

package Adapter;

import java.util.HashMap;
import java.util.Map;

public class AvatarContent {
    /*
         Class which specifies the content of an avatar
     */

    //	the id of the avatar
    public final String AvatarId;

    // The list of MMUs of the session
    public final Map<String, MotionModelUnitBase> MMUs = new HashMap<>();

    //	Basic constructor
    public AvatarContent(String avatarId) {
        this.AvatarId = avatarId;
    }

    //	Returns the MMU based on the id
    MotionModelUnitBase getMMUbyId(String mmuId) throws RuntimeException {
        MotionModelUnitBase result = this.MMUs.get(mmuId);
        if (result != null)
            return result;
        throw new RuntimeException("MMU with id: " + mmuId + " not found in avatar content");
    }
}
